lib/Sema/ConstraintSystem.h

```diff
@@ -2482,7 +2482,8 @@ class ConstraintSystem {
         return false;
 
       // Clearly exponential
-      if (TypeCounter < 32 && CountScopes > (1 << TypeCounter))
+      if (TypeCounter < 32 &&
+        CountScopes >(static_cast<unsigned>(1) << TypeCounter))
```

Comment:
This is what clang format produced, odd. Anyways, I've used 1U now so this is simpler